AllReadyApp/Web-App/AllReady/Areas/Admin/Features/Campaigns/CampaignDetailQueryHandler.cs

```diff
@@ -82,5 +90,13 @@ public CampaignDetailQueryHandler(AllReadyContext context)
 
             return result;
         }
+
+        private CampaignDetailViewModel.CampaignManagerInviteStatus GetCampaignManagerInviteStatus(CampaignManagerInvite campaignManagerInvite)
+        {
+            if (campaignManagerInvite.IsAccepted) return CampaignDetailViewModel.CampaignManagerInviteStatus.Accepted;
+            else if (campaignManagerInvite.IsPending) return CampaignDetailViewModel.CampaignManagerInviteStatus.Pending;
+            else if (campaignManagerInvite.IsRejected) return CampaignDetailViewModel.CampaignManagerInviteStatus.Rejected;
+            else  return CampaignDetailViewModel.CampaignManagerInviteStatus.Revoked;
+        }
```

Comment:
Hi @mgmccarthy - Digging my mind back and having had a quick look at the code I think I know why it was built like that. We're actually storing dates against each status occuring in the db. So the date that the invite was created, date it was accepted etc. I can't recall exactly why now, but I think it stemmed from a discussion I'd had around there needed to be a way to view this history for an invite.

The booleans for the statuses then use the dates to interpret whether those states are in effect. They are not mapped to the db. I think it was a quick thing at the time as we hadn't stubbed out all of the required functionality and I wasn't sure where/when we'd use that info.

I can't think of a driving reason to leave it like that. If we can simplify the code and use for consumers, now would be a good time. There's no reason we couldn't have an enum version and load the appropriate value into that enum based on which status dates have been meet.